### ISSUE TITLE
Upload a base64 image to S3 from client canvas using slingshot

### ISSUE THREAD

**Issue by Billybobbonnet** [CulturalMe/meteor-slingshot#114]
I was wondering how you would proceed to send an image drawn into a client canvas? In a regular case, I would store it as dataURL or send it to blob and convert it to a file server-side, but since i use slingshot it should be exclusively from client to S3, I don't know how to create a file client-side (or even if it feasible). 

I assume that this issue has been already dealt with with slingshot and I would like to know if there is an advised solution for it.

Thanks!

EDIT: here are other stack overflow questions related to this
http://stackoverflow.com/questions/23557848/upload-base64-image-data-to-amazon-s3?rq=1
http://stackoverflow.com/questions/7511321/uploading-base64-encoded-image-to-amazon-s3-via-node-js

**Comment by gsuess** [comment]
Whats the issue?

Just upload the blob... 
```JavaScript

canvas.toBlob(function(blob) {
  uploader.send(blob, function(error, downloadUrl) {/*...*/}) ;
});
```

**Comment by Billybobbonnet** [comment]
Thanks for the answer! I wasn't expecting this to work. I will try it tomorrow.

**Comment by Billybobbonnet** [comment]
My uploads are working correctly with files objects.

I tried to upload the blob directly as you advised. It seems that [`canvas.toBlob` isn't supported everywhere] (http://stackoverflow.com/questions/6736023/which-browsers-and-versions-support-the-canvas-toblob-method) so I used this: https://github.com/blueimp/JavaScript-Canvas-to-Blob

When I upload using the code below, I get a 500 error (no xhr response in the error, xhr isn't defined so I commented it out)
``` javascript
imageCanvas.toBlob(function(blob) {
	uploader.send(blob, function (error, downloadUrl) {
		if (error) {
			// Log service detailed response.
			//console.error('Error uploading', uploader.xhr.response);
			alert (error);
		}
		else {
			console.log(downloadUrl);
		}
	});
});
```
I assume something is going wrong on my side, otherwise I would have a proper `uploader.xhr.response`, but my code is pretty straightforward: the blob is correctly generated and the rest of the code is working with files.
